src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -72,13 +72,67 @@ def _determine_rgw_addr():
         raise LookupError('No RGW daemon found.')
 
     addr = daemon['addr'].split(':')[0]
-    match = re.search(r'port=(\d+)', daemon['metadata']['frontend_config#0'])
-    if match:
-        port = int(match.group(1))
-    else:
-        raise LookupError('Failed to determine RGW port')
+    port, ssl = _parse_frontend_config(daemon['metadata']['frontend_config#0'])
+
+    return addr, port, ssl
+
+
+def _parse_frontend_config(config):
+    """
+    Get the port the RGW is running on. Due the complexity of the
+    syntax not all variations are supported.
+
+    Get more details about the configuration syntax here:
+    http://docs.ceph.com/docs/master/radosgw/frontends/
+    https://civetweb.github.io/civetweb/UserManual.html
+
+    >>> _parse_frontend_config('beast port=8000')
+    (8000, False)
+
+    >>> _parse_frontend_config('civetweb port=8000s')
+    (8000, True)
+
+    >>> _parse_frontend_config('beast port=192.0.2.3:80')
+    (80, False)
 
-    return addr, port
+    >>> _parse_frontend_config('civetweb port=172.5.2.51:8080s')
+    (8080, True)
+
+    >>> _parse_frontend_config('civetweb port=[::]:8080')
+    (8080, False)
```

Comment:
How about using unit tests instead? As they would be up to date with development changes.

Reply:
this documentation is being tested.